addon/-private/system/model/internal-model.js

```diff
@@ -119,7 +119,10 @@ export default class InternalModel {
   constructor(modelName, id, store, data) {
     heimdall.increment(new_InternalModel);
     this.id = id;
-    this._internalId = InternalModelReferenceId++;
+
+    // this ensure ordered set can quickly identify this as unique
+    this[Ember.GUID_KEY] = InternalModelReferenceId++ + 'internal-model';
```

Comment:
Was originally added as part of making stuff faster: https://github.com/emberjs/data/commit/ca9393fb01f23038eb15c479c56e02be4a466558

Left a comment to hope it survives